examples/DA_paper/uphill/uphill.py

```diff
@@ -0,0 +1,221 @@
+import mlptrain as mlt
+import numpy as np
+import matplotlib.pyplot as plt
+from mlptrain.log import logger
+import random
+from mlptrain.box import Box
+from scipy.spatial import distance_matrix
+import os
+import math
+from ase.constraints import Hookean
+from ase.geometry import find_mic
+from generate_rs import generate_rs
+
+mlt.Config.n_cores = 10
+
+def grid_box (positions, size, grid_space):
+    # to put grid box in the system
+    size_x_min = np.min(positions.T[0])-size
+    size_x_max = np.max(positions.T[0])+size
+    size_y_min = np.min(positions.T[1])-size
+    size_y_max = np.max(positions.T[1])+size
+    size_z_min = np.min(positions.T[2])-size
+    size_z_max = np.max(positions.T[2])+size
+
+    x_space = np.arange(size_x_min, size_x_max, grid_space)
+    y_space = np.arange(size_y_min, size_y_max, grid_space)
+    z_space = np.arange(size_z_min, size_z_max, grid_space)
+
+    grid_points = np.vstack(np.meshgrid(x_space,y_space,z_space)).reshape(3,-1).T
+    return grid_points
+
+def cavity_volnum (ase_system, solute_idx = 22, grid_side_length = 0.2, radius = 1.5):
```

Comment:
typo?

Reply:
Marked as resolved but unsure if volnum is a typo for volume (as given in the comment below the function) or not?

Reply:
Still lots of spaces between arguments in function - please run this through Black or another editor as these will resolve all these style issues